nexus/db-model/src/migration.rs

```diff
@@ -0,0 +1,80 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use super::Generation;
+use crate::schema::migration;
+use crate::MigrationState;
+use chrono::DateTime;
+use chrono::Utc;
+use omicron_common::api::internal::nexus;
+use serde::Deserialize;
+use serde::Serialize;
+use uuid::Uuid;
+
+/// The state of a migration as understood by Nexus.
+#[derive(
+    Clone, Debug, Queryable, Insertable, Selectable, Serialize, Deserialize,
+)]
+#[diesel(table_name = migration)]
+pub struct Migration {
+    /// The migration's UUID.
+    ///
+    /// This is the primary key of the migration table and is referenced by the
+    /// `instance` table's `migration_id` field.
+    pub id: Uuid,
+
+    /// The time at which this migration record was created.
+    pub time_created: DateTime<Utc>,
+
+    /// The time at which this migration record was deleted,
+    pub time_deleted: Option<DateTime<Utc>>,
+
+    /// The time at which the source VMM state was last updated.
+
+    /// The state of the migration source VMM.
+    pub source_state: MigrationState,
+
+    /// The ID of the migration source VMM.
+    pub source_propolis_id: Uuid,
+
+    /// The generation number for the source state.
+    pub source_gen: Generation,
+
+    /// The time the source VMM state was most recently updated.
+    pub time_source_updated: Option<DateTime<Utc>>,
+
+    /// The state of the migration target VMM.
+    pub target_state: MigrationState,
```

Comment:
I'm not opposed to starting with the "state" being the same at source and destination, but I wonder if we'll eventually want different types here. Seems possible that there's a source-only or target-only state in the future.

(No change needed, just musing)

Reply:
Right now, the intention for this table is to store pretty low-resolution data about the migration state, basically just whether it's completed or failed, so we're not including more detailed migration status reported by Propolis. So, we don't *really* expect these to include states that are specific to one side of the migration currently, although I suppose we *could*...